src/webgl/p5.DataVector.js

```diff
@@ -0,0 +1,94 @@
+import p5 from '../core/main';
+
+/**
+ * An internal class to store data that will be sent to a p5.RenderBuffer.
+ * Those need to eventually go into a Float32Array, so this class provides a
+ * variable-length array container backed by a Float32Array so that it can be
+ * sent to the GPU without allocating a new array each frame.
+ *
+ * Like a C++ vector, its fixed-length Float32Array backing its contents will
+ * double in size when it goes over its capacity.
+ */
+p5.DataVector = class DataVector {
```

Comment:
I know you're referencing c++ with the vector name, but I feel like within the context of p5 it means something different. Is there a clearer name for this class? DataArray?